src/Servers/Kestrel/Transport.Quic/src/Internal/KestrelServerLimitsFeature.cs

```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.Internal
+{
+    internal class KestrelServerLimitsFeature
```

Comment:
public class QuickListenerConfigFeature

Reply:
Instead of introducing a new public type, what about registering KestrelServerOptions or KestrelServerLimits directly as a feature similar to what we're already doing with SslServerAuthenticationOptions?

Reply:
The dependency is the wrong direction, the transport can't consume KestrelServerOptions. That also seems like it would break the intended layering of transports.

Reply:
Good point. I feel like the better layering is for us to explicitly close the connection if there have been no active streams. If we want to continue exposing QuicTransportOptions.IdleTimeout fine, but I don't think we have to because it is redundant with the KeepAliveTimeout and the shorter one would win.

I get that in some sense the transport can be "better" at avoiding races where a client start the stream right when the server decides to close the connection, but that race exists regardless and the goaway logic should allow the client to safely retry any request for a stream we don't accept.

Reply:
We have `QuicTransportOptions.IdleTimeout` because it's a mandatory setting for System.Net.Quic. We could set a non-configurable value if we wanted, but keeping it aligned to Kestrel's  KeepAliveTimeout seemed preferable.

Reply:
If we manage the lifetime ourselves, I think setting the IdleTimeout to TimeSpan.MaxValue is reasonable. We need to be able to gracefully close connections on-demand for shutdown anyway, and it shouldn't be too difficult to keep track of idleness like we do for HTTP/2.

Reply:
Is that something we want to tackle for .NET 6? Or should we do something simpler like use a hardcoded timeout? Mainly we're trying to avoid having an API on QuicTransportOptions that will be obsolete next version.

Reply:
I think a hardcoded timeout is probably fine for .NET 6. I don't love adding public API that we won't need in the future though.

Reply:
Ok, so remove QuicTransportOptions.IdleTimeout, hard code a value to something conservative like 10 minutes, do not add the KestrelServerLimitsFeature, and update the issue to manage connection lifetimes at the kestrel layer in .NET 7?

Reply:
That seems reasonable to me - what about `MaxUnidirectionalStreamCount` / `MaxBidirectionalStreamCount` - should we still move those to `Http3Limits` & pass them as a feature?